src/main/java/com/microsoft/azure/functions/worker/Application.java

```diff
@@ -122,6 +123,9 @@ public static void main(String[] args) {
         } else {
             try (JavaWorkerClient client = new JavaWorkerClient(app)) {
                 client.listen(app.getWorkerId(), app.getRequestId()).get();
+            } catch (GracefulTerminationException ex) {
+                WorkerLogManager.getSystemLogger().log(Level.INFO, "Gracefully shutting down the worker.");
+                System.exit(0);
```

Comment:
before the exit, we map need to close the rpc channel and also maybe flush the log buffer.